Build fails on x86

@osamu620 The newly relesed 0.28.0 no longer builds on x86, here's the MinGW output:
```
  FAILED: [code=1] src/core/CMakeFiles/openjph.dir/coding/ojph_block_decoder_avx2.cpp.obj 
  D:\M\msys64\mingw32\bin\g++.exe -D_FILE_OFFSET_BITS=64 -ID:/_/B/src/OpenJPH-0.28.0/src/core/openjph -march=pentium4 -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -Wp,-D__USE_MINGW_ANSI_STDIO=1 -O3 -DNDEBUG -std=gnu++14 -fexceptions -Wall -Wextra -Wconversion -Wunused-parameter -mavx2 -MD -MT src/core/CMakeFiles/openjph.dir/coding/ojph_block_decoder_avx2.cpp.obj -MF src\core\CMakeFiles\openjph.dir\coding\ojph_block_decoder_avx2.cpp.obj.d -o src/core/CMakeFiles/openjph.dir/coding/ojph_block_decoder_avx2.cpp.obj -c D:/_/B/src/OpenJPH-0.28.0/src/core/coding/ojph_block_decoder_avx2.cpp
  D:/_/B/src/OpenJPH-0.28.0/src/core/coding/ojph_block_decoder_avx2.cpp: In function 'ojph::ui32 ojph::local::destuff_rev(const ojph::ui8*, int, bool, ojph::ui64, ojph::ui32, ojph::ui8*, ojph::ui32)':
  D:/_/B/src/OpenJPH-0.28.0/src/core/coding/ojph_block_decoder_avx2.cpp:532:25: error: '_mm_cvtsi128_si64' was not declared in this scope; did you mean '_mm_cvtsi32_si64'?
    532 |         ui64 v0 = (ui64)_mm_cvtsi128_si64(r);
        |                         ^~~~~~~~~~~~~~~~~
        |                         _mm_cvtsi32_si64
  D:/_/B/src/OpenJPH-0.28.0/src/core/coding/ojph_block_decoder_avx2.cpp:533:25: error: '_mm_extract_epi64' was not declared in this scope; did you mean '_mm_extract_epi8'?
    533 |         ui64 v1 = (ui64)_mm_extract_epi64(r, 1);
        |                         ^~~~~~~~~~~~~~~~~
        |                         _mm_extract_epi8
```
May I suggest adding this architecture to your CI, or officially dropping support for it otherwise?

> [@osamu620](https://github.com/osamu620) The newly relesed 0.28.0 no longer builds on x86, here's the MinGW output:
> 
> ```
>   FAILED: [code=1] src/core/CMakeFiles/openjph.dir/coding/ojph_block_decoder_avx2.cpp.obj 
>   D:\M\msys64\mingw32\bin\g++.exe -D_FILE_OFFSET_BITS=64 -ID:/_/B/src/OpenJPH-0.28.0/src/core/openjph -march=pentium4 -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -Wp,-D__USE_MINGW_ANSI_STDIO=1 -O3 -DNDEBUG -std=gnu++14 -fexceptions -Wall -Wextra -Wconversion -Wunused-parameter -mavx2 -MD -MT src/core/CMakeFiles/openjph.dir/coding/ojph_block_decoder_avx2.cpp.obj -MF src\core\CMakeFiles\openjph.dir\coding\ojph_block_decoder_avx2.cpp.obj.d -o src/core/CMakeFiles/openjph.dir/coding/ojph_block_decoder_avx2.cpp.obj -c D:/_/B/src/OpenJPH-0.28.0/src/core/coding/ojph_block_decoder_avx2.cpp
>   D:/_/B/src/OpenJPH-0.28.0/src/core/coding/ojph_block_decoder_avx2.cpp: In function 'ojph::ui32 ojph::local::destuff_rev(const ojph::ui8*, int, bool, ojph::ui64, ojph::ui32, ojph::ui8*, ojph::ui32)':
>   D:/_/B/src/OpenJPH-0.28.0/src/core/coding/ojph_block_decoder_avx2.cpp:532:25: error: '_mm_cvtsi128_si64' was not declared in this scope; did you mean '_mm_cvtsi32_si64'?
>     532 |         ui64 v0 = (ui64)_mm_cvtsi128_si64(r);
>         |                         ^~~~~~~~~~~~~~~~~
>         |                         _mm_cvtsi32_si64
>   D:/_/B/src/OpenJPH-0.28.0/src/core/coding/ojph_block_decoder_avx2.cpp:533:25: error: '_mm_extract_epi64' was not declared in this scope; did you mean '_mm_extract_epi8'?
>     533 |         ui64 v1 = (ui64)_mm_extract_epi64(r, 1);
>         |                         ^~~~~~~~~~~~~~~~~
>         |                         _mm_extract_epi8
> ```
> 
> May I suggest adding this architecture to your CI, or officially dropping support for it otherwise?

Thanks for the report, and sorry for the breakage — this is from the decoder rework in #276.

The new destuffing fast path in ojph_block_decoder_avx2.cpp uses two intrinsics that only exist when targeting x86-64 (`_mm_cvtsi128_si64`, `_mm_extract_epi64`). The AVX2 encoder from the same PR already has an i386 fallback for exactly this situation (`proc_cq2` in `ojph_block_encoder_avx2.cpp`); the decoder is the one place that missed the guard. I've checked the tree for other 64-bit-only intrinsics — these two lines are the only occurrence, so the fix is local. Note this is a compile-time issue only: at runtime the AVX2 path is gated by CPU dispatch, so a `-march=pentium4` build never executes it on hardware without AVX2.

I've opened #281 with the fix: on i386 the two 64-bit halves are assembled from 32-bit extracts (`_mm_extract_epi32` is always available in a `-mavx2` TU), and the x86-64 codegen is unchanged. I don't have a 32-bit MinGW toolchain at hand, so confirmation that 0.28.0 + that patch builds in your environment would be much appreciated.

As for adding 32-bit x86 to CI versus officially dropping support — I'll leave that call to @aous72.

--- a/src/core/coding/ojph_block_decoder_avx2.cpp
+++ b/src/core/coding/ojph_block_decoder_avx2.cpp
@@ -529,8 +529,15 @@ namespace ojph {
         __m128i r = _mm_shuffle_epi8(v,
             _mm_set_epi8(0, 1, 2, 3, 4, 5, 6, 7,
                          8, 9, 10, 11, 12, 13, 14, 15)); // reverse bytes
+#ifdef OJPH_ARCH_X86_64
         ui64 v0 = (ui64)_mm_cvtsi128_si64(r);
         ui64 v1 = (ui64)_mm_extract_epi64(r, 1);
+#else // 32-bit x86 lacks the 64-bit extract intrinsics
+        ui64 v0 = (ui32)_mm_cvtsi128_si32(r)
+                | ((ui64)(ui32)_mm_extract_epi32(r, 1) << 32);
+        ui64 v1 = (ui32)_mm_extract_epi32(r, 2)
+                | ((ui64)(ui32)_mm_extract_epi32(r, 3) << 32);
+#endif
         ui64 w0 = acc | (v0 << nb);
         ui64 w1 = (v1 << nb) | (nb ? (v0 >> (64 - nb)) : 0);
         memcpy(o, &w0, 8);

